### PR TITLE
Single float tolerance in comparison fix

### PR DESCRIPTION
**Pull request by maciejgrzybek**
Single floats (32b) have higher machine epsilon value. We should use higher tolerance while comparing single float than we compare doubles.
This patch applies different comparator for floats and for doubles.

### CODE REVIEW

File: tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java [hunk 1]
```diff
@@ -80,7 +82,8 @@ public int compare(Object actual, Object expected)
             case BIGINT:
                 return longEqual(actual, expected);
             case REAL:
-            case FLOAT:
+            case FLOAT: // float may be represented as a 8B but on some DBMSes it's 4B, so it's safer to assume higher tolerance
+                return singleFloatingEqual(actual, expected);
```
Comment (kokosing) [inline tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java]: Remove comment

File: tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java [hunk 2]
```diff
@@ -103,28 +106,29 @@ public int compare(Object actual, Object expected)
...
     {
-        if (actual instanceof Array && expected instanceof List) {
-            Array actualArray = (Array) actual;
-            QueryResultValueComparator elementComparator;
-            elementComparator = comparatorForArrayElements(actualArray);
-            List actualList = arrayAsList(actualArray);
-            List expectedList = (List) expected;
+        if (!(actual instanceof Array && expected instanceof List)) {
+            return -1;
+        }
```
Comment (kokosing) [inline tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java]: This belongs to differnt commit I think

File: tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java [hunk 3]
```diff
@@ -145,7 +149,7 @@ private List arrayAsList(Array array)
             return Arrays.asList((Object[])array.getArray());
         }
         catch (SQLException e) {
-            throw new RuntimeException(e);
```
Comment (kokosing) [inline tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java]: Ditto

**Comment by kokosing** [comment]
Lgtm % comments, I did a review via phone, sorry for mistakes

File: tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java (hunk 3 above)
Comment (maciejgrzybek) [inline tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java]: Extracted to a separate commit (730e1fc5d64e4eccd557efb1cb29d825918470b1).

File: tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java (hunk 2 above)
Comment (maciejgrzybek) [inline tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java]: Extracted to a separate commit (730e1fc5d64e4eccd557efb1cb29d825918470b1).

File: tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java (hunk 1 above)
Comment (maciejgrzybek) [inline tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java]: Done. Commit (24f929d79f485e55861f6a0909847a8f896f3a75) amended.

**Comment by maciejgrzybek** [comment]
Thanks and no worries. Fortunately it is a small fix. :)

**Comment by maciejgrzybek** [comment]
After "LGTM" on Slack, merged to master. Thanks!
